quint/src/ir/namespacer.ts

```diff
@@ -63,6 +63,27 @@ class Namespacer implements IRTransformer {
     return expr
   }
 
+  exitTuple(tup: QuintTup): QuintTup {
+    // Process each element of the tuple
+    const namespacedElements = tup.elements.map(element => {
+      // If the element is a name, apply the namespace
+      if (element.kind === 'name' && typeof element.name === 'string') {
+        return {
+          ...element,
+          name: namespacedName(this.namespace, element.name), // Apply the namespace
+        }
+      }
+      // Otherwise, return the element as-is
+      return element
+    })
+
+    // Return a new tuple with namespaced elements
+    return {
+      ...tup,
+      elements: namespacedElements,
+    }
+  }
+
```

Comment:
We actually don't need this! The IR transformer itself should already guarantee that we visited/transformed all of the elements before reaching the tuple. This works as it is right now because `namespacedName` is idempotent (it can be applied more than once and the result will be the same as if it was applied once, that is, `f(x) = f(f(x))`), but it will also work as if we remove this, and we should :smile: 
```suggestion
```